### PR TITLE
fix(docs): generate d.ts file only for top level modules.

### PR DESCRIPTION
**Pull request by rkirov**

### CODE REVIEW

File: docs/dgeni-package/index.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ module.exports = new Package('angular', [jsdocPackage, nunjucksPackage, linksPac
   ];
 
   readTypeScriptModules.sourceFiles = [
-    '*/*.@(js|es6|ts])',
```
Comment (rkirov) [inline docs/dgeni-package/index.js]: I spent quite some time digging through to figure out why were were only picking up .ts files :)

Comment (petebacondarwin) [inline docs/dgeni-package/index.js]: This is a good fix. Thansk

File: docs/dgeni-package/processors/createTypeDefinitionFile.js [hunk 2]
```diff
@@ -14,11 +14,12 @@ module.exports = function createTypeDefinitionFile() {
         modules: []
       };
       _.forEach(docs, function(doc) {
-        if ( doc.docType === 'module' ) {
+        var isTopLevelModule = /angular2\/[^\/]*$/.test(doc.id);
+        if ( doc.docType === 'module' && isTopLevelModule) {
```
Comment (petebacondarwin) [inline docs/dgeni-package/processors/createTypeDefinitionFile.js]: I don't think this is the correct fix. If you are only interested in top level modules then should we not be using the `public_docs` dgeni package, which only loads the top level docs in the first place. See https://github.com/angular/angular/blob/master/docs/public-docs-package/index.js#L7

Comment (rkirov) [inline docs/dgeni-package/processors/createTypeDefinitionFile.js]: Actually, I stand corrected, I only need `angular2/angular2` as we don't want to push the complexity of understanding our module structure to the users of the `.d.ts`. They should just do `import {angular2/angular2}`.

**Comment by vsavkin** [comment]
@rkirov what is the status of this PR? Could you assign an action?

**Pull request closed without merging**